dynamic_array.c

```diff
@@ -34,15 +36,16 @@ void buffer_destroy(struct buffer *inst)
 // If realloc SUCCEDS we modify the size
 static void try_realloc(struct buffer *inst, size_t new_capacity)
 {
-    inst->buffer = realloc_zero(inst->buffer, 
-                                buffer_size(inst) * sizeof(void*),
-                                new_capacity * sizeof(void*));
+    inst->buffer = buffer_realloc_zero(inst->buffer, 
+                                       buffer_size(inst) * sizeof(void*),
+                                       new_capacity * sizeof(void*));
     for (size_t i = inst->capacity; i < new_capacity; ++i) {
         void *succeds =  realloc(inst->buffer[i], inst->item_size);
         if (succeds){
             inst->buffer[i] = succeds;
         } else {
-            exit(0);
+            log_error("%s\n", "Realloc failed!");
```

Comment:
At some point, we will switch over to using a memory manager which I wrote for our first satellite, which does a few cool error-detecting things.  That memory manager will automatically halt the program if a malloc/realloc fails.  But we can keep these checks in for now since we are not using that memory manager yet.